api/proto/teleport/identitycenter/v1/identitycenter.proto

```diff
@@ -0,0 +1,201 @@
+// Copyright 2024 Gravitational, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+syntax = "proto3";
+
+package teleport.identitycenter.v1;
+
+import "teleport/header/v1/resourceheader.proto";
+
+option go_package = "github.com/gravitational/teleport/api/gen/proto/go/teleport/identitycenter/v1;identitycenterv1";
+
+// PermissionSetInfo represents a permission set that is available on a given
+// account. This is not intended to be a complete description of a PermissionSet,
+// but just enough to identify it for cross-referencing when included in an
+// Account record.
+message PermissionSetInfo {
+  // Arn is the unique ARN for this permission set
+  string arn = 1;
+
+  // Title is the human-readable name of the permission set
+  string name = 2;
+
+  // Role is an optional ARN indicating role provisioned to this account. May be
+  // empty if the permission set is not provisioned or is not relevant in the
+  string role = 3;
+}
+
+// AccountStatus represents any commonly-changing data about an Identity
+// Center account.
+message AccountStatus {}
+
+// Account is an Identity-Ceneter-managed AWS account
+message Account {
+  teleport.header.v1.ResourceHeader header = 1;
+  AccountSpec spec = 2;
+  AccountStatus status = 3;
```

Comment:
After spending some time today in our dev branch, I found we have few instances of `*.Status.Status` usage which was pretty confusing without following through each of them. 
If we know the status ends up stacking up with other resources, what do you think about naming these status fields  in full `account_status`, `assignment_status`?

(I am usually proponent of just using `status` otherwise but it is a loss here)

Reply:
The name of the toplevel status block is dictated by RFD 153, so that has to stay.  

WRT to the fields under `status` - I'm open to them being more descriptive. In some cases I think I have conflated the idea of `Status` (the current collection of data about the resource) and `State` (which particular part of the resource lifecycle state machine the resource is in). 

I'll have a look and see if I can clarify stuff.

Reply:
Noted on the RFD 153 status field, and yeah, double checking now and its more on the provisioning side that we abuse `Status.Status`.

Reply:
See: #47424